src/model-helpers/Profile.js

```diff
@@ -0,0 +1,160 @@
+  const profileDataset = await getWebIdDataset(session.info.webId);
+  const profileThing = getThing(profileDataset, session.info.webId);
+
+  const profileName = getStringNoLocale(profileThing, RDF_PREDICATES.profileName);
+  const nickname = getStringNoLocale(profileThing, RDF_PREDICATES.nickname);
+  const profileImage = getUrl(profileThing, RDF_PREDICATES.profileImg);
+
+  const profileInfo = { profileName, nickname, profileImage };
+
+  // TODO: include more fields to the object like organization, address, etc.
+  // when expanding this feature
+  return { profileInfo, profileDataset, profileThing };
+};
+
+/**
+ * A function that updates the user's profile information based on what is inputed
+ * in the Profile page
+ *
+ * @function updateProfileInfo
+ * @param {Session} session - Solid's Session Object {@link Session}
+ * @param {object} profileData - The object containing the information related
+ * to the person on their profile card, the profile dataset, and the profile Thing
+ * @param {object} inputValues - The inputs for updating profile information
+ * on their profile card
+ * @returns {Promise} Promise - Performs action to update profile card on the
+ * user's profile card
+ */
+export const updateProfileInfo = async (session, profileData, inputValues) => {
+  let { profileDataset, profileThing } = profileData;
+  const { profileInfo } = profileData;
+
+  Object.keys(inputValues).forEach((input) => {
+    switch (inputValues[input]) {
+      case null:
+        return;
+      case '':
+        profileThing = removeStringNoLocale(
+          profileThing,
+          RDF_PREDICATES[input],
+          profileInfo[input]
+        );
+        break;
+      default:
+        profileThing = buildThing(profileThing)
+          .setStringNoLocale(RDF_PREDICATES[input], inputValues[input])
+          .build();
+        break;
+    }
+  });
+
+  profileDataset = setThing(profileDataset, profileThing);
+
+  await saveSolidDatasetAt(session.info.webId, profileDataset, { fetch: session.fetch });
+};
+
+/**
+ * Function that upload profile image of user onto their profile container and
+ * reference it on their Pod
+ *
+ * @function uploadProfileImage
+ * @param {Session} session - Solid's Session Object {@link Session}
+ * @param {object} profileData - The object containing the information related
+ * to the person on their profile card, the profile dataset, and the profile Thing
+ * @param {Blob} inputImage - File blob being uploaded for profile image
+ * @returns {Promise} Promise - Performs a file upload of your profile image onto
+ * your pod at /profile
+ */
+export const uploadProfileImage = async (session, profileData, inputImage) => {
+  let { profileDataset, profileThing } = profileData;
+
+  const profileContainer = `${session.info.webId.split('card')[0]}`;
+  const profileImgFilename = inputImage.name.replace(' ', '%20');
+
+  await saveFileInContainer(profileContainer, inputImage, {
+    slug: profileImgFilename,
```

Comment:
Image upload is failing on solid community. It seems that node solid server does not fully respect the slug suggestion here. If I upload a file name `tim.jpg`, node solid server stores it as `tim.jpg.jpeg`.

![image](https://github.com/codeforpdx/PASS/assets/37914436/d041ec2f-1651-44f9-879d-5203f31fd7ac)


I believe the `saveFileInContainer` function should return the actual url of the created file. We should be able to store that on the profile instead.

Reply:
Found the bug. This happened because I've turned on `contentType` for `saveFileInContainer` from `solid-client`. If I kept `contentType: inputImage.type` it seems to lump `.jpg` and `.jpeg` together.

I'll have this removed and have the function imply what the file is. Seemed to have fixed this error.

https://github.com/codeforpdx/PASS/assets/14917816/a6356ad2-6563-4bde-b07c-67819f5ab453

Reply:
Nevermind, I still see the problem on solidcommunity.net. I'll see if I can patch this again.

Reply:
Seems to be a weird quirk with extensions that has 4- and 3-letter counterparts. For `tiff` files, it'll come out as `<filename>.tiff.tif`, similar to how `.jpg` is returning `.jpg.jpeg`.

Was able to fix the `.jpg` issue when uploading by converting all `.jpg` into `.jpeg`. The same cannot be said for `.tiff` to `.tif` unfortunately. 

Think we can drop `.tiff` though since we're already supporting `.png`, `.jpg`, `.jpeg`, `.gif`, `.webp` and `.tiff` seems to only be supported by Safari browsers (https://developer.mozilla.org/en-US/docs/Web/Media/Formats/Image_types). Gonna base most of what we accept as image files from the common image file types shown in that mdn link.

Reply:
Your current solution requires individual mappings for every individual file type to whatever abbreviation node solid server uses. That doesn't seem very maintainable to me. What if we find another server that saves files as `jpg` instead of `jpeg`? What if we get a server that ignores the slug suggestion altogether? Then this code breaks.

The `saveFileInContainer` function returns the URL the server assigned to the file. Just save that URL into the profile dataset. Then you don't have to worry about any of this filetype mapping. There are examples of how to do this on Inrupt's website: https://docs.inrupt.com/developer-tools/javascript/client-libraries/tutorial/read-write-files/#save-a-file-into-an-existing-container

Reply:
I'm a bit confused by this. I believe I was following the example from Inrupt for `saveFileInContainer`. Do you mean the portion about the `getSourceUrl`? I don't think I've tried that before. I'll give it a go.

Reply:
Think I got this updated to using the source URL from `saveFileInContainer` using a new constant `savedFile`. Created a new function in session-helper to help with generating the associated URL for the Thing in the TTL file.

Reply:
Yeah this looks good